test/stdlib/Net_HTTP_test.rb

```
@@ -1,50 +1,105 @@
-    assert_send_type "(String, String) -> nil",
-                     Net::HTTP, :get_print, 'www.ruby-lang.org', '/en'
-    assert_send_type "(URI::Generic, Hash[String, String]) -> nil",
-                     Net::HTTP, :get_print, URI("https://www.ruby-lang.org"), {"Accept" => "text/html"}
-    assert_send_type "(URI::Generic, Hash[Symbol, String]) -> nil",
-                     Net::HTTP, :get_print, URI("https://www.ruby-lang.org"), {Accept: "text/html"}
-    assert_send_type "(URI::Generic) -> String",
-                     Net::HTTP, :get, URI("https://www.ruby-lang.org")
-    assert_send_type "(String, String) -> String",
-                     Net::HTTP, :get, 'www.ruby-lang.org', '/en'
-    assert_send_type "(URI::Generic, Hash[String, String]) -> String",
-                     Net::HTTP, :get, URI("https://www.ruby-lang.org"), {"Accept" => "text/html"}
-    assert_send_type "(URI::Generic, Hash[Symbol, String]) -> String",
-                     Net::HTTP, :get, URI("https://www.ruby-lang.org"), {Accept: "text/html"}
-    assert_send_type "(URI::Generic) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, URI("https://www.ruby-lang.org")
-    assert_send_type "(String, String) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, 'www.ruby-lang.org', '/en'
-    assert_send_type "(URI::Generic, Hash[String, String]) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, URI("https://www.ruby-lang.org"), {"Accept" => "text/html"}
-    assert_send_type "(URI::Generic, Hash[Symbol, String]) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, URI("https://www.ruby-lang.org"), {Accept: "text/html"}
+    with_server("localhost") do |uri|
+      assert_send_type "(URI::Generic) -> nil",
+                       Net::HTTP, :get_print, uri
+      assert_send_type "(String, String) -> nil",
+                       Net::HTTP, :get_print, "www.ruby-lang.org", "/en"
```

Some lines look like they still connect to `ruby-lang.org`. Is this intentional?

---

I was testing while disconnecting the PC from the network, but it seems the error was being suppressed in the `rescue` block.